File: src/main/java/com/extendedae_plus/mixin/bridge/HelperProviderMenu.java
package com.extendedae_plus.mixin.bridge;

import appeng.api.inventories.InternalInventory;
import appeng.helpers.patternprovider.PatternProviderLogic;
import net.pedroksl.advanced_ae.common.logic.AdvPatternProviderLogic;

public interface HelperProviderMenu {
    default PatternProviderLogic getProviderLogicVanilla() {
        return null;
    }

    default AdvPatternProviderLogic getProviderLogicAdv() {
        return null;
    }

    default InternalInventory getInvPattern() {
        var logicVanilla = this.getProviderLogicVanilla();
        var logicAdv = this.getProviderLogicAdv();
        if (logicVanilla != null) return logicVanilla.getPatternInv();
        else return logicAdv.getPatternInv();
    }
}

File: src/main/kotlin/com/extendedae_plus/mixin/helper/HelperProviderMenu.kt
package com.extendedae_plus.mixin.helper

import appeng.api.inventories.InternalInventory
import appeng.helpers.patternprovider.PatternProviderLogic
import net.pedroksl.advanced_ae.common.logic.AdvPatternProviderLogic

interface HelperProviderMenu {
    val providerLogicVanilla: PatternProviderLogic?
        get() = null

    val providerLogicAdv: AdvPatternProviderLogic?
        get() = null

    val invPattern: InternalInventory?
        get() {
            val logicVanilla = this.providerLogicVanilla
            val logicAdv = this.providerLogicAdv
            return logicVanilla?.patternInv ?: logicAdv?.patternInv
        }
}
